### ISSUE TITLE
react/require-optimization warns when using PureRenderMixin.shouldComponentUpdate

### ISSUE THREAD

**Issue by hmhealey** [yannickcr/eslint-plugin-react#1019]
`react/require-optimization` reports the following as incorrect even though it follows Facebook's documentation for using PureRenderMixin with ES6 classes ([link](https://facebook.github.io/react/docs/pure-render-mixin.html)):
```javascript
export class MyComponent extends React.Component {
    constructor(props) {
        super(props);
        this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
    }
}
```

I can try submitting a fix if you agree that this is actually a bug

**Comment by yannickcr** [comment]
Currently `PureRenderMixin` is only supported on ES5 class.

If you can submit a fix I'll be happy to merge it 😃

**Comment by ljharb** [comment]
@hmhealey why not `extends React.PureComponent` at that point then?

**Comment by hmhealey** [comment]
@ljharb Wow, I did not know that existed. That's a lot nicer. Thanks!

I'll still try to take a look at fixing this if I get a chance. I'm curious to learn a bit about the internals of ESLint plugins.

**Comment by hmhealey** [comment]
I spent some time looking at this over the weekend and judging by the code for the `constructor-super` rule, it would require 200-300 lines of code to check to make sure that the constructor or its parent sets `this.shouldComponentUpdate` at all times. That seems like a lot just to support an outdated way of creating pure components, so I don't think this needs to be fixed right now.
